zfs-replicate.sh

```diff
@@ -363,37 +363,53 @@ do_snap() {
             printf "Creating ZFS snapshot %s@%s\n" "${local_set}" "${sname}"
             if [ $RECURSE_CHILDREN -ne 1 ]; then
                 printf "RUNNING: %s snapshot %s@%s\n" "${ZFS}" "${local_set}" "${sname}"
-                $ZFS snapshot ${local_set}@${sname}
+                if ! $ZFS snapshot ${local_set}@${sname}; then
```

Comment:
I think adding some sanity checks are good, but what's more readable and specific? Just checking for bash truthy/falsy or being specific about the error code with `if [[ $? -ne 0 ]]` for example. I think the latter is more clear.

Reply:
Where would you include that line?
Its better to exit when failing any command, and outputting the error is also good.

Reply:
It would look something like this ...

```bash
$ZFS snapshot ${local_set}@${sname}
if [[ $? -ne 0 ]]; then
  exit_error
fi
```

In Bash `$?` is the exit-code of the previous command. We could also modify `exit_error` to optionally take an argument so we could do something like ...
```bash
$ZFS snapshot ${local_set}@${sname}
if [[ $? -ne 0 ]]; then
  exit_error "ZFS snapshot failed, command exited: $?"
fi
```

Reply:
Ive included it now.